lib/api/CFieldDataCategorizer.cc

```diff
@@ -113,12 +105,12 @@ bool CFieldDataCategorizer::handleRecord(const TStrStrUMap& dataRowFields, TOpti
 
     CGlobalCategoryId globalCategoryId{this->computeAndUpdateCategory(dataRowFields, time)};
     if (globalCategoryId.isHardFailure() == false) {
-        m_OutputFieldCategory =
-            core::CStringUtils::typeToString(globalCategoryId.globalId());
-        if (m_OutputHandler.writeRow(dataRowFields, m_Overrides) == false) {
-            LOG_ERROR(<< "Unable to write output with type " << m_OutputFieldCategory
-                      << " for input:" << core_t::LINE_ENDING
-                      << this->debugPrintRecord(dataRowFields));
+        if (m_OutputFieldCategory != nullptr) {
```

Comment:
It shouldn't ever be null for `autodetect`.  In `autodetect` the categorizer will always be chained to an anomaly detector.

We also have the `categorize` program though, where the categorizer is completely standalone and isn't chained to anything.  In this case it doesn't need to pass on its result in the input record.  It just needs to write it to the `CJsonOutputWriter`, which is higher up the file.

The `categorize` program is not plumbed into the Java, as we decided part way through the pure categorization work that it wasn't worth doing because people could just do a `count by mlcategory` detector and ignore the anomalies if they wanted pure categorization.  But the `categorize` program was already written.  It would be a shame to delete it and then find there _was_ a worthwhile use case for it.  The one difference is that `categorize` can operate on data without a time field whereas `autodetect` requires a time field.